src/Policeman/Download/Hackage.hs

```diff
@@ -0,0 +1,53 @@
+{- | Functions to download Haskell packages from Hackage.
+-}
+
+module Policeman.Download.Hackage
+    ( PackageName (..)
+    , downloadFromHackage
+    ) where
+
+import Control.Exception (catch, throwIO)
+import Shellmet (($?))
+import System.Directory (createDirectoryIfMissing, removeDirectoryRecursive)
+import System.FilePath ((</>))
+import System.IO.Error (IOError, isDoesNotExistError)
+
+import Policeman.Core.Package (PackageName (..), Version (..))
+import Policeman.Download.Common (DownloadError (..), evidenceDir)
+
+
+{- | This function takes 'PackageName' and previous package
+'Version', downloads @.tar.gz@ archive from Hackage and unpacks it in
+the current directory.
+-}
```

Comment:
Could we always check just the latest one? Or you think it could be useful to be able to compare between older versions?

Well, actually now I think that this could be cool. However, how would we know the desired "previous version"?

Reply:
Yeah, that's a bit challenging. I've created an issue for that:

* https://github.com/kowainik/policeman/issues/12